src/bin/load_test.rs

```diff
@@ -199,39 +236,170 @@ async fn main() {
     )
     .await;
 }
 
+async fn broadcast_contract_variants(
+    sdk: Arc<Sdk>,
+    identity: &Identity,
+    signer: Arc<SimpleSigner>,
+    data_contract: &DataContract,
+    count: u32,
+    seed: u64,
+) -> Vec<DataContract> {
+    tracing::info!("registering data contracts");
+    let mut rng = StdRng::seed_from_u64(seed);
+    let data_contract_variants = (0..count)
+        .into_iter()
+        .map(|i| {
+            let entropy: [u8; 32] = rng.gen();
+
+            let new_id = DataContract::generate_data_contract_id_v0(identity.id(), entropy);
+
+            let mut data_contract_variant = data_contract.clone();
+            data_contract_variant.set_id(new_id);
+            CreatedDataContract::from_contract_and_entropy(
+                data_contract_variant,
+                entropy.into(),
+                PlatformVersion::latest(),
+            )
+            .expect("expected to get contract")
+        })
+        .collect::<Vec<_>>();
+
+    let partial_identity = identity.clone().into_partial_identity_info();
+
+    let key_to_use = identity
+        .get_first_public_key_matching(
+            Purpose::AUTHENTICATION,
+            HashSet::from([SecurityLevel::CRITICAL]),
+            HashSet::from([KeyType::ECDSA_SECP256K1]),
+        )
+        .expect("expected to get a key");
+
+    let mut transitions_queue = VecDeque::new();
+
+    let first_exists = DataContract::fetch(
+        &sdk,
+        data_contract_variants.first().unwrap().data_contract().id(),
+    )
+    .await
+    .expect("expected to get data contract")
+    .is_some();
+
+    for data_contract_variant in data_contract_variants.iter().rev() {
+        let exists = if !first_exists {
+            false
+        } else {
+            DataContract::fetch(&sdk, data_contract_variant.data_contract().id())
+                .await
+                .expect("expected to get data contract")
+                .is_some()
+        };
+        if !exists {
+            let transition = DataContractCreateTransition::new_from_data_contract(
+                data_contract_variant.data_contract().clone(),
+                *data_contract_variant.entropy_used(),
+                &partial_identity,
+                key_to_use.id(),
+                signer.as_ref(),
+                PlatformVersion::latest(),
+                None,
+            )
+            .expect("expected transition");
+
+            transitions_queue.push_front(transition); // we push to the front on purpose
+        } else {
+            break; // if one exists we can assume all previous ones also exist
+        }
+    }
+
+    for (i, transaction) in transitions_queue.iter().enumerate() {
+        tracing::info!("registering contract {}", i);
+        if i % 24 == 0 {
+            transaction
+                .broadcast_and_wait(&sdk, None)
+                .await
+                .expect("expected to register contract");
+        } else {
+            transaction
+                .broadcast(&sdk)
```

Comment:
If we just broadcast, we might not detect some issues, right?

If so, maybe use broadcast_and_wait but DON'T await immediately, just gather join handles and await for them all at the end? For example, with tokio::join! or tokio::task::JoinSet

Reply:
Yes the test would fail, but then the next time we run the test things would solve themselves. Again this is not meant to be production level code. I just continued what was already here.